src/lib/web-component/log-view/log-view-element.js

```diff
@@ -0,0 +1,87 @@
+class LogView extends HTMLElement {
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    this.logTableTemplate = document.querySelector('.log-table-template');
+    this.logTableRowTemplate = document.querySelector(
+      '.log-table-row-template'
+    );
+
+    const logTableInstance = document.importNode(
+      this.logTableTemplate.content,
+      true
+    );
+
+    this.logDetailWrapper = logTableInstance.querySelector(
+      '.log-detail-wrapper'
+    );
+    this.logDetails = logTableInstance.querySelector('#logDetails');
+    this.closeBtn = logTableInstance.querySelector('.close');
+    this.logTableWrapper = logTableInstance.querySelector('.log-table-wrapper');
+    this.tableBody = logTableInstance.querySelector('tbody');
+
+    shadow.appendChild(logTableInstance);
+  }
+
+  addNewRows(logs) {
+    for (const log of logs) {
+      const logTableRowInstance = document.importNode(
+        this.logTableRowTemplate.content,
+        true
+      );
+      logTableRowInstance.querySelector('tr')._log = log;
```

Comment:
Ideally you would assign directly to `logTableRowInstance`. Assuming that the template contains one element only, you could achieve the desired goal by using `this.logTableRowTemplate.content.firstElementChild`.

And instead of reading that property so often here, you could directly read that from the constructor, so you can simply do `const logTableRowInstance = this.logTableRowTemplate.cloneNode(true);` here. (cloneNode and `document.importNode` are equivalent, but cloneNode is shorter).

Reply:
done :-)